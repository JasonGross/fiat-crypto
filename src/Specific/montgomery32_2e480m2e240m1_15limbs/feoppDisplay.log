opp = λ var : Z.Syntax.base_type → Type,
λ '(x26, x27, x25, x23, x21, x19, x17, x15, x13, x11, x9, x7, x5, x3, x1)%core,
uint32_t x29, uint8_t x30 = subborrow_u32(0x0, 0x0, x1);
uint32_t x32, uint8_t x33 = subborrow_u32(x30, 0x0, x3);
uint32_t x35, uint8_t x36 = subborrow_u32(x33, 0x0, x5);
uint32_t x38, uint8_t x39 = subborrow_u32(x36, 0x0, x7);
uint32_t x41, uint8_t x42 = subborrow_u32(x39, 0x0, x9);
uint32_t x44, uint8_t x45 = subborrow_u32(x42, 0x0, x11);
uint32_t x47, uint8_t x48 = subborrow_u32(x45, 0x0, x13);
uint32_t x50, uint8_t x51 = subborrow_u32(x48, 0x0, x15);
uint32_t x53, uint8_t x54 = subborrow_u32(x51, 0x0, x17);
uint32_t x56, uint8_t x57 = subborrow_u32(x54, 0x0, x19);
uint32_t x59, uint8_t x60 = subborrow_u32(x57, 0x0, x21);
uint32_t x62, uint8_t x63 = subborrow_u32(x60, 0x0, x23);
uint32_t x65, uint8_t x66 = subborrow_u32(x63, 0x0, x25);
uint32_t x68, uint8_t x69 = subborrow_u32(x66, 0x0, x27);
uint32_t x71, uint8_t x72 = subborrow_u32(x69, 0x0, x26);
uint32_t x73 = cmovznz32(x72, 0x0, 0xffffffff);
uint32_t x75, uint8_t x76 = addcarryx_u32(0x0, x29, (x73 & 0xffffffff));
uint32_t x78, uint8_t x79 = addcarryx_u32(x76, x32, (x73 & 0xffffffff));
uint32_t x81, uint8_t x82 = addcarryx_u32(x79, x35, (x73 & 0xffffffff));
uint32_t x84, uint8_t x85 = addcarryx_u32(x82, x38, (x73 & 0xffffffff));
uint32_t x87, uint8_t x88 = addcarryx_u32(x85, x41, (x73 & 0xffffffff));
uint32_t x90, uint8_t x91 = addcarryx_u32(x88, x44, (x73 & 0xffffffff));
uint32_t x93, uint8_t x94 = addcarryx_u32(x91, x47, (x73 & 0xffffffff));
uint32_t x96, uint8_t x97 = addcarryx_u32(x94, x50, (x73 & 0xfffeffff));
uint32_t x99, uint8_t x100 = addcarryx_u32(x97, x53, (x73 & 0xffffffff));
uint32_t x102, uint8_t x103 = addcarryx_u32(x100, x56, (x73 & 0xffffffff));
uint32_t x105, uint8_t x106 = addcarryx_u32(x103, x59, (x73 & 0xffffffff));
uint32_t x108, uint8_t x109 = addcarryx_u32(x106, x62, (x73 & 0xffffffff));
uint32_t x111, uint8_t x112 = addcarryx_u32(x109, x65, (x73 & 0xffffffff));
uint32_t x114, uint8_t x115 = addcarryx_u32(x112, x68, (x73 & 0xffffffff));
uint32_t x117, uint8_t _ = addcarryx_u32(x115, x71, (x73 & 0xffffffff));
(Return x117, Return x114, Return x111, Return x108, Return x105, Return x102, Return x99, Return x96, Return x93, Return x90, Return x87, Return x84, Return x81, Return x78, Return x75)
     : ∀ var : Z.Syntax.base_type → Type, expr Z.Syntax.base_type Z.Syntax.op (uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t -> uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t)

Argument scope is [function_scope]
