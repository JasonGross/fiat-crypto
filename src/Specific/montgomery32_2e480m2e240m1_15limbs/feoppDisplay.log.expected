λ x : word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32,
Interp-η
(λ var : Syntax.base_type → Type,
 λ '(x27, x28, x26, x24, x22, x20, x18, x16, x14, x12, x10, x8, x6, x4, x2)%core,
 uint32_t x30, uint8_t/*bool*/ x31 = subborrow_u32(0x0, 0x0, x2);
 uint32_t x33, uint8_t/*bool*/ x34 = subborrow_u32(x31, 0x0, x4);
 uint32_t x36, uint8_t/*bool*/ x37 = subborrow_u32(x34, 0x0, x6);
 uint32_t x39, uint8_t/*bool*/ x40 = subborrow_u32(x37, 0x0, x8);
 uint32_t x42, uint8_t/*bool*/ x43 = subborrow_u32(x40, 0x0, x10);
 uint32_t x45, uint8_t/*bool*/ x46 = subborrow_u32(x43, 0x0, x12);
 uint32_t x48, uint8_t/*bool*/ x49 = subborrow_u32(x46, 0x0, x14);
 uint32_t x51, uint8_t/*bool*/ x52 = subborrow_u32(x49, 0x0, x16);
 uint32_t x54, uint8_t/*bool*/ x55 = subborrow_u32(x52, 0x0, x18);
 uint32_t x57, uint8_t/*bool*/ x58 = subborrow_u32(x55, 0x0, x20);
 uint32_t x60, uint8_t/*bool*/ x61 = subborrow_u32(x58, 0x0, x22);
 uint32_t x63, uint8_t/*bool*/ x64 = subborrow_u32(x61, 0x0, x24);
 uint32_t x66, uint8_t/*bool*/ x67 = subborrow_u32(x64, 0x0, x26);
 uint32_t x69, uint8_t/*bool*/ x70 = subborrow_u32(x67, 0x0, x28);
 uint32_t x72, uint8_t/*bool*/ x73 = subborrow_u32(x70, 0x0, x27);
 uint32_t x74 = cmovznz32(x73, 0x0, 0xffffffff);
 uint32_t x75 = (x74 & 0xffffffff);
 uint32_t x77, uint8_t/*bool*/ x78 = addcarryx_u32(0x0, x30, x75);
 uint32_t x79 = (x74 & 0xffffffff);
 uint32_t x81, uint8_t/*bool*/ x82 = addcarryx_u32(x78, x33, x79);
 uint32_t x83 = (x74 & 0xffffffff);
 uint32_t x85, uint8_t/*bool*/ x86 = addcarryx_u32(x82, x36, x83);
 uint32_t x87 = (x74 & 0xffffffff);
 uint32_t x89, uint8_t/*bool*/ x90 = addcarryx_u32(x86, x39, x87);
 uint32_t x91 = (x74 & 0xffffffff);
 uint32_t x93, uint8_t/*bool*/ x94 = addcarryx_u32(x90, x42, x91);
 uint32_t x95 = (x74 & 0xffffffff);
 uint32_t x97, uint8_t/*bool*/ x98 = addcarryx_u32(x94, x45, x95);
 uint32_t x99 = (x74 & 0xffffffff);
 uint32_t x101, uint8_t/*bool*/ x102 = addcarryx_u32(x98, x48, x99);
 uint32_t x103 = (x74 & 0xfffeffff);
 uint32_t x105, uint8_t/*bool*/ x106 = addcarryx_u32(x102, x51, x103);
 uint32_t x107 = (x74 & 0xffffffff);
 uint32_t x109, uint8_t/*bool*/ x110 = addcarryx_u32(x106, x54, x107);
 uint32_t x111 = (x74 & 0xffffffff);
 uint32_t x113, uint8_t/*bool*/ x114 = addcarryx_u32(x110, x57, x111);
 uint32_t x115 = (x74 & 0xffffffff);
 uint32_t x117, uint8_t/*bool*/ x118 = addcarryx_u32(x114, x60, x115);
 uint32_t x119 = (x74 & 0xffffffff);
 uint32_t x121, uint8_t/*bool*/ x122 = addcarryx_u32(x118, x63, x119);
 uint32_t x123 = (x74 & 0xffffffff);
 uint32_t x125, uint8_t/*bool*/ x126 = addcarryx_u32(x122, x66, x123);
 uint32_t x127 = (x74 & 0xffffffff);
 uint32_t x129, uint8_t/*bool*/ x130 = addcarryx_u32(x126, x69, x127);
 uint32_t x131 = (x74 & 0xffffffff);
 uint32_t x133, uint8_t/*bool*/ _ = addcarryx_u32(x130, x72, x131);
 (Return x133, Return x129, Return x125, Return x121, Return x117, Return x113, Return x109, Return x105, Return x101, Return x97, Return x93, Return x89, Return x85, Return x81, Return x77))
x
     : word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 → ReturnType (uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t)
